### PR TITLE
fix(tracing) handling tracestate header correctly

### PR DESCRIPTION
**Pull request by shettyh**
### Summary

When `tracestate` header is multi valued ( it is allowed by the spec ) zipkin plugin was failing with error ```bad argument #1 to 'match' (string expected, got table)```

### Full changelog

* [zipkin] handling multi value tracestate header correctly

### CODE REVIEW

File: kong/tracing/propagation.lua [hunk 1]
```diff
@@ -343,6 +343,15 @@ local function find_header_type(headers)
   local b3_single_header = headers["b3"]
   if not b3_single_header then
     local tracestate_header = headers["tracestate"]
+
+    -- handling tracestate header if it is multi valued
+    if tracestate_header and type(tracestate_header) == "table" then
```
Comment (mayocream) [inline kong/tracing/propagation.lua:348]: nit:
```suggestion
    if type(tracestate_header) == "table" then
```

File: kong/tracing/propagation.lua [hunk 2]
```diff
@@ -343,6 +343,15 @@ local function find_header_type(headers)
...
+      -- https://www.w3.org/TR/trace-context/#tracestate-header
+      -- Handling multi value header : https://httpwg.org/specs/rfc7230.html#field.order
+      tracestate_header = table.concat(tracestate_header, ',')
```
Comment (mayocream) [inline kong/tracing/propagation.lua:351]: localize `table.concat`

File: kong/tracing/propagation.lua [hunk 3]
```diff
@@ -343,6 +343,15 @@ local function find_header_type(headers)
...
+      kong.log.warn("Trace state header is table : " .. tracestate_header)
```
Comment (mayocream) [inline kong/tracing/propagation.lua:352]: If it's expected behavior, then should not log a warn level msg.
```suggestion
      kong.log.debug("header `tracestate` is a table")
```

File: spec/01-unit/26-tracing/02-propagation_spec.lua [hunk 4]
```diff
@@ -149,6 +149,13 @@ describe("propagation.parse", function()
       assert.spy(warn).not_called()
     end)
 
+    it("multi value tracestate header", function()
+      local tracestate_header = { "test", trace_id, span_id }
+      local t = { parse({ tracestate =  tracestate_header }) }
+      assert.same({ }, to_hex_ids(t))
+      assert.spy(warn).called(1)
```
Comment (mayocream) [inline spec/01-unit/26-tracing/02-propagation_spec.lua:156]:
```suggestion
      assert.spy(debug).called(1)
```

File: kong/tracing/propagation.lua (hunk 1 above)
Comment (shettyh) [reply kong/tracing/propagation.lua]: Updated

File: kong/tracing/propagation.lua (hunk 2 above)
Comment (shettyh) [reply kong/tracing/propagation.lua]: Done

File: kong/tracing/propagation.lua (hunk 3 above)
Comment (shettyh) [reply kong/tracing/propagation.lua]: done

File: spec/01-unit/26-tracing/02-propagation_spec.lua (hunk 4 above)
Comment (shettyh) [reply spec/01-unit/26-tracing/02-propagation_spec.lua]: done

**Review by mayocream — APPROVED** [review]
Thanks for the quick response!

**Pull request merged**